### PR TITLE
feat(explorer): cost per proof chart

### PR DESCRIPTION
**Pull request by MarcosNicolau**
## Description
- Integrate [chart.js](https://www.chartjs.org/) library and create `line_chart` component.
- Fetch `fee_per_proof` for the last `n` `batches`
- Add a new module in `home` that shows the cost per proof of the last `100` batches

Screenshot 2024-12-19 at 2 48 35 PM

**Testing**
- Start a local devnet.
- Run the explorer.
- Send proofs and see how the graph starts to fill. 

## Type of change

- [x] New feature

## Checklist

- [ ] “Hotfix” to `testnet`, everything else to `staging`
- [ ] Linked to Github Issue
- [ ] This change depends on code or research by an external entity
  - [ ] Acknowledgements were updated to give credit
- [ ] Unit tests added
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] This change is an Optimization
  - [ ] Benchmarks added/run
- [ ] Has a known issue
  - [Link to the open issue addressing it]() 
- [ ] If your PR changes the Operator compatibility (Ex: Upgrade prover versions)
  - [ ] This PR adds compatibility for operator for both versions and do not change batcher/docs/examples
  - [ ] This PR updates batcher and docs/examples to the newer version. This requires the operator are already updated to be compatible

### CODE REVIEW

**Review by PatStiles — APPROVED** [review]

File: explorer/lib/explorer_web/components/charts.ex
Comment (uri-99) [inline explorer/lib/explorer_web/components/charts.ex:1]: The chart is missing its axes

File: explorer/lib/explorer_web/live/eth_converter.ex [hunk 1]
```diff
@@ -40,7 +40,6 @@ defmodule EthConverter do
       {:ok, Decimal.round(usd_value, decimal_places) |> Decimal.to_string(:normal)}
     else
       {:error, reason} -> {:error, reason}
-      _ -> {:error, "Failed to convert wei to USD"}
```
Comment (uri-99) [inline explorer/lib/explorer_web/live/eth_converter.ex:43]: why was this removed?

**Review by uri-99 — CHANGES_REQUESTED** [review]
I like this tooltip, we could add more info to it.
amount of proofs, merkle_root, gas per proof and gas_price.
Also "block number" could be better with "submition block number" for more clarity
image

File: explorer/lib/explorer_web/live/eth_converter.ex (hunk 1 above)
Comment (MarcosNicolau) [reply explorer/lib/explorer_web/live/eth_converter.ex:43]: It is redundant the {:error, reason} already accounts for it

File: explorer/lib/explorer_web/components/charts.ex
Comment (MarcosNicolau) [reply explorer/lib/explorer_web/components/charts.ex:1]: Yeah, that is on purpose, it is what we have on the design. Also, axes make it look very bad honestly.

File: explorer/lib/explorer_web/live/pages/home/index.ex [hunk 2]
```diff
@@ -1,7 +1,23 @@
...
 
+  def get_cost_per_proof_chart_data() do
+    data = Batches.get_fee_per_proofs_of_last_n_batches(100)
+
+    %{
+      data:
+        Enum.map(data, fn {fee_per_proof, _} ->
+          case EthConverter.wei_to_usd(fee_per_proof) do
+            {:ok, value} -> value
+            {:error, _} -> 0
```
Comment (uri-99) [inline explorer/lib/explorer_web/live/pages/home/index.ex]: we had this `error -> 0` once before and it is not a good idea. a nil is better, you can then avoid drawing this point

Comment (MarcosNicolau) [reply explorer/lib/explorer_web/live/pages/home/index.ex]: I am afraid `nil` might break the liveview somewhere and throw a 500

Comment (MarcosNicolau) [reply explorer/lib/explorer_web/live/pages/home/index.ex]: I guess a better alternative is to filter them

File: explorer/lib/explorer_web/live/pages/home/index.ex [hunk 3]
```diff
@@ -1,7 +1,23 @@
...
+  import ExplorerWeb.ChartComponents
   use ExplorerWeb, :live_view
 
+  def get_cost_per_proof_chart_data() do
+    data = Batches.get_fee_per_proofs_of_last_n_batches(100)
+
+    %{
+      data:
+        Enum.map(data, fn {fee_per_proof, _} ->
+          case EthConverter.wei_to_usd(fee_per_proof) do
```
Comment (uri-99) [inline explorer/lib/explorer_web/live/pages/home/index.ex]: also here you are missing the second param in wei_to_usd, which says how many decimals you want. i suggest 2.

File: explorer/lib/explorer_web/components/charts.ex
Comment (MarcosNicolau) [reply explorer/lib/explorer_web/components/charts.ex:1]: Solved in [d89d3b0](https://github.com/yetanotherco/aligned_layer/pull/1635/commits/d89d3b0351099483eb896d082f3d66342d3dd070)

File: explorer/lib/explorer_web/live/pages/home/index.ex (hunk 2 above)
Comment (MarcosNicolau) [reply explorer/lib/explorer_web/live/pages/home/index.ex]: Solved in [7165b55](https://github.com/yetanotherco/aligned_layer/pull/1635/commits/7165b5526d079dc6f84e86ac3f0b7552ca18fc28)

File: explorer/lib/explorer_web/live/pages/home/index.ex (hunk 3 above)
Comment (MarcosNicolau) [reply explorer/lib/explorer_web/live/pages/home/index.ex]: Solved in [fabdab0](https://github.com/yetanotherco/aligned_layer/pull/1635/commits/fabdab044aacaf7e6b1b190f4fdf6c8cb26c7495)

**Review by uri-99 — APPROVED** [review]

**Review by uri-99 — CHANGES_REQUESTED** [review]
X axis is not proportional, all proofs are equidistant even though they may have different time between them

File: explorer/lib/explorer_web/live/pages/home/index.ex [hunk 4]
```diff
@@ -34,8 +66,7 @@ defmodule ExplorerWeb.Home.Index do
 
     operators_registered = Operators.get_amount_of_operators()
 
-    latest_batches =
-      Batches.get_latest_batches(%{amount: 10})
+    latest_batches = Batches.get_latest_batches(%{amount: 5, order_by: :desc})
```
Comment (uri-99) [inline explorer/lib/explorer_web/live/pages/home/index.ex]:
```suggestion
    latest_batches = Batches.get_latest_batches(%{amount: 10, order_by: :desc})
```

**Review by uri-99 — APPROVED** [review]
